codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/RestXmlProtocolGenerator.java

```diff
@@ -0,0 +1,122 @@
+package software.amazon.smithy.aws.go.codegen;
+
+import java.util.Set;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.go.codegen.GoStackStepMiddlewareGenerator;
+import software.amazon.smithy.go.codegen.GoWriter;
+import software.amazon.smithy.go.codegen.integration.HttpBindingProtocolGenerator;
+import software.amazon.smithy.go.codegen.integration.HttpProtocolTestGenerator;
+import software.amazon.smithy.go.codegen.integration.HttpProtocolUnitTestRequestGenerator;
+import software.amazon.smithy.go.codegen.integration.HttpProtocolUnitTestResponseErrorGenerator;
+import software.amazon.smithy.go.codegen.integration.HttpProtocolUnitTestResponseGenerator;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.MemberShape;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.traits.TimestampFormatTrait;
+
+abstract class RestXmlProtocolGenerator extends HttpBindingProtocolGenerator {
+    /**
+     * Creates a AWS REST XML protocol generator.
+     */
+    RestXmlProtocolGenerator() {
+        super(true);
+    }
+
+    @Override
+    public void generateProtocolTests(GenerationContext context) {
+        new HttpProtocolTestGenerator(context,
+                new HttpProtocolUnitTestRequestGenerator.Builder(),
+                new HttpProtocolUnitTestResponseGenerator.Builder(),
+                new HttpProtocolUnitTestResponseErrorGenerator.Builder()
+        ).generateProtocolTests();
+    }
+
+    @Override
+    protected TimestampFormatTrait.Format getDocumentTimestampFormat() {
+        return null;
+    }
+
+    @Override
+    protected String getDocumentContentType() {
+        return null;
+    }
+
+    @Override
+    protected void generateOperationDocumentSerializer(
+            GenerationContext context, OperationShape operation
+    ) {
+
+    }
+
+    @Override
+    protected void writeMiddlewareDocumentSerializerDelegator(
+            Model model, SymbolProvider symbolProvider, OperationShape operation,
+            GoStackStepMiddlewareGenerator generator, GoWriter writer
+    ) {
+
+    }
+
+    @Override
+    protected void writeMiddlewareErrorDeserializer(
+            GoWriter writer,
+            Model model,
+            SymbolProvider symbolProvider,
+            OperationShape operationShape,
+            GoStackStepMiddlewareGenerator generator
+    ) {
+        writer.openBlock("if response.StatusCode < 200 || response.StatusCode >= 300 {", "}", () -> {
+            writer.write("// TODO: Implement error deserializer delegators");
```

Comment:
Even though this is in progress, it would be nice to change this comment to return a simple error with this TODO.